### PR TITLE
Build OpenCV winpack with DLDT 2021.4.0

### PR DESCRIPTION
**Pull request by alalek**
https://github.com/openvinotoolkit/openvino/releases/tag/2021.4

relates #16847 #17056 #17466 #17759 #18534 #19158 #19786

Wiki: https://github.com/opencv/opencv/wiki/Intel%27s-Deep-Learning-Inference-Engine-backend#opencvdldt-windows-package-community-version

Preview packages:
- [3.4 branch](http://pullrequest.opencv.org/buildbot/builders/precommit_custom_windows/builds/2446) ( [+debug](http://pullrequest.opencv.org/buildbot/builders/precommit_custom_windows/builds/2429) )
- [master branch](http://pullrequest.opencv.org/buildbot/builders/precommit_custom_windows/builds/2438) ( [+debug](http://pullrequest.opencv.org/buildbot/builders/precommit_custom_windows/builds/2440) )

<cut/>

```
force_builders=Custom Win
build_image:Custom Win=winpack-dldt
```

### CODE REVIEW

File: platforms/winpack_dldt/2021.4/sysroot.config.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+copy_bin('cache.json')
+copy_dll('clDNNPlugin')
+copy_dll('HeteroPlugin')
+copy_dll('inference_engine')
+copy_dll('inference_engine_ir_reader')
+copy_dll('inference_engine_ir_v7_reader')
+copy_dll('inference_engine_legacy')
+copy_dll('inference_engine_transformations')  # runtime
+copy_dll('inference_engine_lp_transformations')  # runtime
+copy_dll('inference_engine_preproc')  # runtime
```
Comment (alalek) [inline platforms/winpack_dldt/2021.4/sysroot.config.py:27]: @dkurt @ilya-lavrenov Hi, do you know if we need to add `inference_engine_preproc.dll` for inferencing on IE through OpenCV DNN module?

Comment (ilya-lavrenov) [reply platforms/winpack_dldt/2021.4/sysroot.config.py:27]: From what I see in dnn code - we don't trigger IE preprocessing. 

As a dummy check you can run dnn tests and if they don't load preprocessing dynamically, it's not used in opencv. Can be checked via `strace`

Comment (dkurt) [reply platforms/winpack_dldt/2021.4/sysroot.config.py:27]: Yes, dnn does not utilize OpenVINO preprocessing

Comment (alalek) [reply platforms/winpack_dldt/2021.4/sysroot.config.py]: @ilya-lavrenov @dkurt  Thank you!
Removed this .dll from the package (opencv_test_dnn tests pass).

File: platforms/winpack_dldt/2021.4/sysroot.config.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    global copy_bin, dll_suffix
+    copy_bin(name + dll_suffix + '.dll')
+    copy_bin(name + dll_suffix + '.pdb')
+
+copy_bin('cache.json')
+copy_dll('clDNNPlugin')
+copy_dll('HeteroPlugin')
+copy_dll('inference_engine')
+copy_dll('inference_engine_ir_reader')
+copy_dll('inference_engine_ir_v7_reader')
```
Comment (ilya-lavrenov) [inline platforms/winpack_dldt/2021.4/sysroot.config.py:23]: `inference_engine_ir_v7_reader` is deprecated and used only in IE tests, should be removed as well.

**Comment by alalek** [comment]
:+1:

**Pull request merged**